test/claim-aud.test.js

```diff
@@ -3,13 +3,19 @@
 const jwt = require('../');
 const expect = require('chai').expect;
 const util = require('util');
+const testUtils = require('./test-utils');
 
-function signWithAudience(payload, audience) {
+function signWithAudience(audience, payload, callback) {
   const options = {algorithm: 'none'};
   if (audience !== undefined) {
     options.audience = audience;
   }
-  return jwt.sign(payload, undefined, options);
+
+  testUtils.signJWTHelper(payload, 'secret', options, callback);
+}
+
+function verifyWithAudience(token, audience,  callback) {
+  testUtils.verifyJWTHelper(token, undefined, {audience}, callback);
```

Comment:
🤔 Shouldn't this fail because you are passing an undefined secret?

Reply:
Verify is okay with having an `undefined` secret. Only when signing a token is it required because of https://github.com/brianloveswords/node-jws/issues/62 .

Reply:
Also, if you provide a secret when verifying and the algorithm is set to `none`, as they are in theses tests, verify will throw an error.

Reply:
The error would be: ```
JsonWebTokenError: jwt signature is required
```

Reply:
If the token has signature the secret is required:
```
> token = jwt.sign({a:1},'secret')
'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjoxLCJpYXQiOjE1Mzk3MDk0MTR9.R7aZOSiLJvskXR1jDdg4r891cT_OXHVycWDBcZFccFM'
> jwt.verify(token, undefined)
JsonWebTokenError: secret or public key must be provided
    at /Users/ediaz/wkf/auth0/node-jsonwebtoken/verify.js:92:19
```
See it is sync code.

Reply:
Oh, I see, I didn't realize about the `none`.

Reply:
That's because by default the algorithm is set to HS256.

If you sign with `token = jwt.sign({a:1},'secret', {algorithm: 'none'})`:

```
> token = jwt.sign({a:1},'secret', {algorithm: 'none'});
'eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJhIjoxLCJpYXQiOjE1Mzk3MDk2OTd9.'
> jwt.verify(token, 'secret')
JsonWebTokenError: jwt signature is required
    at /home/mitmaro/code/node-jsonwebtoken/verify.js:88:19
    at getSecret (/home/mitmaro/code/node-jsonwebtoken/verify.js:76:14)
    at Object.module.exports [as verify] (/home/mitmaro/code/node-jsonwebtoken/verify.js:80:10)
```